### DISCUSSION TITLE
Apple Watch support

### DISCUSSION THREAD

**Discussion by mshort9999** [creepymonster/GlucoseDirect#466] [Q&A]
Can you tell me where I can get directions for setting up the Apple Watch and Lock Screen?

**Comment by Aixelsyd007** [comment]
I don't know of any instructions other than the README and the FAQ linked within. I was able to get it working yesterday and will leave a guide here for anyone that googles upon this thread. 

First you need to go to:

 Calendar app > Calendars > Add Calendar 

 You can name it anything you like, and I suppose you could create a calendar with another account, but I just went with iCloud

Next open Glucose Direct:

 scroll down to Apple Export Settings, enable Export to Apple Calendar, and enter the name of your new calendar

![8A7ACEF0-F997-4574-AC9D-0FDA28F8D659_1_102_o](https://github.com/creepymonster/GlucoseDirect/assets/16341422/302a95bb-72db-49ed-8d4b-ba15d94c53ec)

Now switch to your Watch (This step can also be performed from your iPhone's Watch app):

# long press the home scr..  wait, no..  3d touch the h...   nope, that's not it either.   🤔   Um,   ..H aPtIc  tOuCh.. maybe??  🤷 whatever they're calling it now 

anywho, Force Push on yer home screen, then Edit on your face

Swipe L/R over to the Complications page:

 there are many Complication shapes and sizes, but the smaller ones aren't going to display the Calendar item's Title and Location which contain the glucose data we want

I've attached some examples below showing layouts that work and ones that basically just display the current time(meeting start time)

![2E29D198-F056-4BEF-A948-BEDD542430F6](https://github.com/creepymonster/GlucoseDirect/assets/16341422/b1514f80-c187-4be3-96b6-c1e5708e0ca3)

![65BE17A9-E737-4199-8F85-7CB50377A7DF_1_102_o](https://github.com/creepymonster/GlucoseDirect/assets/16341422/4a33b46f-d2e4-4571-951e-9c3b546d7f95)

![incoming-F8EFDAB1-B6DD-4DA4-A02D-A91E6533DAE0](https://github.com/creepymonster/GlucoseDirect/assets/16341422/83f3afdb-cf29-4cb4-a3ca-1de9f9eabdfa)

![incoming-4C79FD51-2D88-4AF2-9641-A81374426E70](https://github.com/creepymonster/GlucoseDirect/assets/16341422/f6a85960-0dea-4763-9e72-78908fb413df)

You could use these small ones as shortcut buttons so that you're only one click away from the glucose reading, but I prefer the reading to be displayed on the home screen

That should be all you need to get the watch portion working



You probably already have the first part working, I think those instructions are in the documentation, but to recap: 

(I'm using a Libre 3 sensor with an iPhone 12 running iOS 16.5.1/WatchOS9.5.2) 

Open your primary Libre app ("Libre 3" in my case) > Menu > LibreLinkUp > Manage > Add Connection, then enter your name and the email associated with your account

Now open the LibreLinkUp app and accept the new connection. Data should begin flowing in after a few minutes. You can now enter your LibreLinkUp login info in the Glucose Direct > Settings page. The data is still being gathered by the official app over Bluetooth and then has to pass through the LibreLinkUp app before Glucose Direct can get any readings. I read that the only reason for NFC scanning the sensor within Glucose Direct is to establish a Bluetooth connection with the sensor so that Glucose Direct can prevent that connection from timing out. 

You don't need to enable any of the Glucose Notifications, etc. at the bottom of the Settings > Glucose Settings section in order for the app to store readings or for the watch portion to function. It will still pull new readings and update the Calendar event in the background, but it seems to lag far less if you enable the Glucose Live Activity option. This places a persistent notification on your phone's lock screen and prevents Glucose Direct from going to background/sleep. I've been running with it enabled all day and haven't noticed any real impact to battery life. I'm not using the alarms so I'm not certain about those, but you likely have to enable the Alarm Glucose Notification if you plan to use those.

The Libre 3 and Glucose Direct apps can be running at the same time without issue(in fact it's necessary for any part of this to function), but I've heard that leaving LibreLinkUp or LibreView running in the background can cause issues. It might even still cause issues if logged in to one of those apps from another device. I'm not sure if it's a local issue with those app both running, or if it's simply an error caused by the same LinkUp account being connected to 2 sessions at once

The Export to Apple Health doesn't need to be enabled for the app or the watch complication to function, but I recommend turning it on as it's by far the most solid Health export feature I've tested while trying numerous glucose monitoring apps in efforts to get this data on my watch face. I’ve been looking forward to trying this beta for a couple months now since it was already full when I discovered it. I was able to join a few days ago and it does not disappoint. I am very impressed with Glucose Direct so far.

Entering insulin dosage or blood glucose monitor results manually into Glucose Direct won't copy back in the other direction to the Freestyle apps, but it will be copied forward into Apple Health. I prefer this as I like Glucose Direct’s manual entry interface much better than Freestyle's version. I also already have my other medical records and health/fitness data in Health. If you use LibreView to share your data with your healthcare provider, they will still be able to see the sensor provided glucose readings the same as before, and, if needed, you could easily export all of the insulin intake and finger stick entries along with your sensor data from the Apple Health app and send it to them. I think there is even a similar live sharing option within the Apple Health app that allows you to grant access to your provider. Note: adding insulin dosage and/or manually entering Glucose readings in the Libre 3 app logbook will not copy forwards into Glucose Direct; only the sensor readings do. 




Here's something I ran into recently and it's sort of tangentially related to this topic. I decided to add prescriptions and dosing times into the Medicines section of the Apple Health app. It works quite well to remind you of all medicines to be taken at a certain time of day. You can also quickly mark the group of medicines for that dose time as taken by clicking on the notification > Mark All as Taken. I also like having the medications complication mapped to one of the smaller home screen Watch buttons for quick access. This is the part where it becomes tangentially relative: 

 I finally got all of my medicine info entered into the Health > Medicines section (I take a lot of prescriptions). I later installed the latest iOS and WatchOS updates only to have the Medications Watch Complication option vanish from the Face Gallery setup on both the Watch and the iPhone app. A quick google suggested that the fix was to disconnect and reconnect the Health app from iCloud and then to **ugh**, 🤦... delete all of the medicines from the Health app and manually re-add them. I was determined to avoid that step if at all possible. Fortunately, just doing the first part, Settings > AppleID > iCloud > Show All > Health > "Sync this iPhone" -off > Keep on my iPhone > "Sync this iPhone" -on, was enough to restore the Medicines Complication on both the iPhone and Watch. I also rebooted the phone and the watch at this time for good measure, but that may not be required. 

I figured if it happened to the Medicines Complication during an upgrade then it could possibly happen to one of the other data sources.

**Reply by philipp-1337** [reply]
Amazing guide. Thanks @Aixelsyd007 . What are your experiences with the battery consumption/draining after enabling the calendar export/sync so far (both: iPhone and Apple Watch)?

**Reply by Aixelsyd007** [reply]
Sure thing!

It's hard to say for sure just by looking at the battery monitor instead of doing a proper benchmark test. I was using other apps during this time. Still, in the last 12 hours with Glucose Direct set to update every minute, Health Export, Calendar Export, and Live Monitor enabled; the battery monitor is showing Glucose Direct as being responsible for only 3% of overall power use. Again, that's not a valid test, but it certainly seems to consume less power than I was expecting.


I didn't check the watch usage really, but it seems normal and I wouldn't expect a noticeable impact there since the watch is only responsible for receiving the updated Calendar event. It shouldn't be more than a few lines to process every minute.

**Reply by philipp-1337** [reply]
@Aixelsyd007 thanks. Very helpful. I had the connection enabled in the past, but I experienced a heavy drain. Maybe I had it set up somehow wrong. I'll try it again. 👍🏼

**Reply by Aixelsyd007** [reply]
I'll keep it on for a few days to get a better idea of the impact on battery life. I am seeing more power usage now, Glucose Direct is showing 8%. Also, I didn't it notice before, but the health app is at 20% which seems very high. I guess the Health app export feature is causing the excessive power use on the receiving end.
